zio-stream/src/main/scala/neotypes/zio/stream/ZioStreams.scala

```diff
@@ -1,22 +1,25 @@
 package neotypes.zio.stream
 
-import neotypes.exceptions.CancellationException
+import neotypes.model.exceptions.CancellationException
 
 import zio.{Exit, Task}
 import zio.stream.{ZSink, ZStream}
 import zio.interop.reactivestreams.Adapters
 import org.reactivestreams.FlowAdapters.toPublisher
 
 import java.util.concurrent.Flow.Publisher
-import scala.collection.compat._
+import scala.collection.Factory
 
 trait ZioStreams {
   implicit final val zioStream: neotypes.Stream.Aux[ZioStream, Task] =
     new neotypes.Stream[ZioStream] {
       override final type F[T] = Task[T]
 
-      override final def fromPublisher[A](publisher: Publisher[A]): ZioStream[A] =
-        Adapters.publisherToStream(toPublisher(publisher), bufferSize = 16)
+      override final def fromPublisher[A](publisher: => Publisher[A], chunkSize: Int): ZioStream[A] =
```

Comment:
ChunkSize < 2 will cause an assertion failure in the internals of Zio as it eventually gets passed directly to Zio's RingBuffer impl. Causes a test failure in this branch. Not sure how you want to handle it so just commenting here to save you the trouble.

Reply:
Why I am not surprised of **ZIO** making my life harder than it already is.

There are many singleton `Publishers` in this program and generally in the wild, why does the buffer have to be of size `2`?!

Reply:
You can see the assertion [here](https://github.com/zio/zio/blob/ba58425b23a4beb60b511ce2bb5babafc40de591/core/js/src/main/scala/zio/internal/RingBuffer.scala#L28). It looks like the code was inspired by [this impl](https://www.1024cores.net/home/lock-free-algorithms/queues/bounded-mpmc-queue), which has the same assertion.